### PR TITLE
feat(source): add gh cli local github validation

### PR DESCRIPTION
**Pull request by jonathanhaaswriter**
## Summary
- auto-resolve local GitHub source auth and repo config from the GH CLI when running the CLI without explicit token/owner/repo arguments
- add focused unit coverage for the GH CLI hydration path on source and source-runtime commands
- add a live opt-in end-to-end test that compares Cerebro GitHub source output against `gh api` and projects the result into local Kuzu

## Testing
- make verify
- `go run ./cmd/cerebro source read github per_page=1 state=all`
- `go run ./cmd/cerebro source read github family=audit per_page=1`
- `CEREBRO_RUN_GITHUB_LOCAL_E2E=1 go test ./cmd/cerebro -run TestGitHubLocalEndToEndWithGHCLI -v`

### CODE REVIEW

**Comment by jonathanhaaswriter** [comment]
@droid Please perform a deep code review and security review of this PR. Leave inline comments only for high-confidence, actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review this PR after the latest fixes. Focus on whether the prior code review and security findings are resolved, and call out any remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fixes. Focus on whether the prior findings on this PR are resolved and only report remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
### Droid sweep status

Latest sweep: `2026-04-29 22:09 UTC`.

- Unresolved non-outdated review threads: `0`
- CI `verify`: `SUCCESS`
- `droid-review`: `not present`

I’ll keep watching for new review findings/check failures and address them as they appear.

**Comment by jonathanhaaswriter** [comment]
Re-audited the live unresolved thread (including outdated) against the current branch tip. The live GitHub E2E now re-fetches the exact PR by number before asserting fields, so the separate-request ordering race is already addressed; no code change was needed. Validation run: none locally during this audit; existing branch `verify` is green. Remaining unresolved threads after this audit: 0.

**Comment by jonathanhaaswriter** [comment]
Droid sweep audit complete (final pass)

- Re-scanned all live review threads, including outdated ones
- Verified each remaining thread against the current branch tip
- Resolved every actionable thread (or fixed it inline and pushed)
- 0 unresolved review threads across the whole open-PR stack at this time

PR #494 (fix/feedback-main-followup) and PR #495 (feat/droid-creation-workflows) received additional fixes for SSRF (DNS-rebind) and workflow branch-name validation in this pass.

**Comment by jonathanhaaswriter** [comment]
@droid review

**Comment by jonathanhaaswriter** [comment]
@droid please harden this PR end-to-end:

1. Re-review the diff for residual bugs, races, edge cases, security gaps, and untested paths.
2. Apply the smallest correct fixes for any high-confidence issues you find.
3. Strengthen test coverage: add regression tests for every prior review finding on this PR plus the new behavior, including failure paths, boundary inputs, concurrency, and not-found / typed-nil interface guards.
4. Run focused tests and `make verify`. If validators fail, fix until green.
5. Commit and push the fixes to this PR's branch (or, if you lack push permission in this workflow, leave concrete inline suggestions and a summary comment).

Do not change the PR scope, do not modify the default branch, and do not merge.

**Comment by jonathanhaaswriter** [comment]
@droid review

File: cmd/cerebro/github_local.go [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+		false,
+	)
+	if err != nil {
+		return nil, err
+	}
+	// Preserve a caller-supplied token (an explicit, persisted PAT) but never persist a token we
+	// just hydrated from the local gh CLI session.
+	if _, ok := originalConfig["token"]; !ok {
+		delete(config, "token")
+	}
```
[the remark this answers is not in the document]
Comment (jonathanhaaswriter) [reply cmd/cerebro/github_local.go]: Fixed in e3a902902 by treating empty/whitespace caller tokens as absent before deleting hydrated gh-cli tokens; added a whitespace-token regression.

**Comment by jonathanhaaswriter** [comment]
@droid review

Please re-review the gh-cli-backed local GitHub validation deeply. The auto-resolve path now consumes external CLI state, so the security and correctness boundary matters:

1. **Token sourcing and leakage**: verify resolved tokens from `gh auth token` are kept in-memory only, never logged, never echoed in error messages, never persisted to disk via fixtures/test artifacts, and never serialized into preview payloads.
2. **Subprocess safety**: confirm the gh CLI invocation uses fixed argv (no shell), validates exit codes, applies timeouts, bounds output size, and degrades gracefully when `gh` is missing, unauthenticated, or returns non-JSON.
3. **Owner/repo inference correctness**: ensure inferred owner/repo come from a trusted source (e.g., explicit gh repo view) and cannot be coerced from attacker-controlled remotes/configs in the working tree; reject ambiguous or empty results.
4. **Override precedence**: verify explicit token/owner/repo flags always take precedence over auto-resolve, and that auto-resolve only fires when all three are missing, not partially.
5. **Live e2e opt-in safety**: confirm the new live test only runs when `CEREBRO_RUN_GITHUB_LOCAL_E2E=1`, that it cannot accidentally hit production, that it cleans up local Kuzu artifacts, and that failures don’t leak token contents.
6. **Error taxonomy**: ensure missing/unauthenticated gh, invalid gh output, and network failures map to caller-actionable errors without leaking internal command details or stack traces.
7. **Cross-platform behavior**: re-check macOS/Linux path resolution, PATH lookup, and gh CLI version skew so the auto-resolve doesn’t silently fall through on environments without gh installed.
